src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    } else if (this.length === 1) {
+      return `${this[0]}`;
+    }
+
+    let result = '';
+    let b = 0;
```

Comment:
give a more understandable name to your variable

Reply:
Next time - for sure :) Should I remake this one?